app/javascript/src/components/Dashboard/Contacts/Pane/constants.js

```diff
@@ -0,0 +1,28 @@
+import { buildSelectOption } from "utils";
+import * as yup from "yup";
+
+import { ROLES } from "components/Dashboard/constants";
+
+export const INITIAL_VALUES = {
+  firstName: "",
+  lastName: "",
+  emails: [],
+  role: "",
```

Comment:
Sorry didn't notice.

Reply:
@yedhink I checked my code right now. Those initial values are passed to [`buildContactInitialValues`](https://github.com/vivek-viswam-rv/neeto-ui-challenge/pull/20/files#:~:text=buildContactInitialValues(contact)%7D) function, hence `role: ""` is the correct initial value.😅

Reply:
Then you should NOT name this literal constant as `INITIAL_VALUES` giving the wrong impression to the developer. Name it something like `DUMMY_CONTACT`.

Reply:
Also I don't get why you need a `[buildContactInitialValues](https://github.com/vivek-viswam-rv/neeto-ui-challenge/pull/20/files#:~:text=buildContactInitialValues(contact)%7D)` and on top of that you require a `INITIAL_VALUES ` constant too.

Either ways, the yup validations should be done on top of the return value of `buildContactInitialValues`, or whichever value has `role` as `null` rather than `empty string`. Make the necessary changes to fix this review comment.

Reply:
[`buildContactInitialValues`](https://github.com/vivek-viswam-rv/neeto-ui-challenge/pull/20/files#:~:text=export%20const-,buildContactInitialValues,-%3D%20contact%20%3D%3E) is used to build `{label: ..., value:...}` etc, irrespective of whether the data is an initial value or stored contact value. 

I think renaming it to `EMPTY_CONTACT` would resolve this confusion. `INITIAL_VALUE` was not supposed to mean initial values for the form, it was supposed to be an empty contact. Shall I rename it to `EMPTY_CONTACT`?

Also I will modify the `buildContactInitialValues` to return `role` as `null` if the input `role` is "".

Would the above changes suffice🤔? Could you please share your thoughts on this?

Reply:
@vivek-viswam-rv https://www.loom.com/share/d7d205b11b8e40c686a5ddbd3d3bdd71

Reply:
Yedin, please check out this video 
https://www.loom.com/share/9a1a67dcbfb44a9199347512219d65b4

The above video shows the latest commit. `INITIAL_VALUES` has been renamed to `EMPTY_CONTACT`.
Also I will modify the `EMPTY_CONTACT.role` to `null` and use short-circuiting instead of the ternary in the `buildContactInitialValues`.

Reply:
@vivek-viswam-rv I understand what the issue is. But the Form's initial values should be a separate constant. In actual codebase, we get the `contact` resource value from the Backend. And thus it will be React state and we pass that state to `Edit` component's `Form`, rather than `INITIAL_VALUES`. So here create two constants. One is `FORM_INITIAL_VALUES`. Another one is `DUMMY_CONTACT`. Use `FORM_INITIAL_VALUES` for `Create`. Use `DUMMY_CONTACT` for `Edit`.

PS: The `buildContactInitialValues` is only useful right now given that you don't have any backend data. Once you have the backend data, all necessary values for a `contact`, say like `emails` etc, will be fetched from backend and thus you won't have to run `buildContactInitialValues` in the case where we are invoking the `Form` from `Edit` component.

Reply:
I will modify the code accordingly. Thanks for your input, I really appreciate your suggestions.😀